code/ATMOSPHERICS/components/binary_devices/pump.dm

```diff
@@ -237,12 +237,7 @@ Thus, the two variables affect pump operation are set in New():
 
 /obj/machinery/atmospherics/binary/pump/attackby(obj/item/W, mob/user, params)
 	if(istype(W, /obj/item/pen))
-		var/t = copytext(stripped_input(user, "Enter the name for the pump.", "Rename", name), 1, MAX_NAME_LEN)
-		if(!t)
-			return
-		if(!in_range(src, usr) && loc != usr)
-			return
-		name = t
+		rename_interactive(user, W)
```

Comment:
In the interest of transparency, I should probably note that this (and other atmos machinery) are technically a behavior change, since I'm now forcing the new name to be "gas pump - N2O", with "gas pump" being mandatory.
The old behavior, where you could name the pump just "N2O" can be easily achieved by passing in `use_prefix = FALSE`, but I don't think this is actually needed.